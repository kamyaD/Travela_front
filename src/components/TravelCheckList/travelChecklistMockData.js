const date = new Date();
const departureDate = new Date(date.setDate(date.getDate() + 1))
  .toISOString().split('T')[0];
const returnDate = new Date(date.setDate(date.getDate() + 3))
  .toISOString().split('T')[0];

export const travelChecklists = [
  {
    'destination': 'Kampala, Uganda',
    'checklist':[
      {'id': '1',
        'name': 'Visa Application',
        'resources': [{
          'checklistItemId':1,
          'id':2,
          'label': 'Application guide',
          'link': 'https://google.com/application-guide'
        },
        ],
        'requiresFiles':true,
        'deleteReason':null,
      },{
        'id': 2,
        'name': 'Ticket Information',
        'resources': [{
          'checklistItemId':2,
          'id':1,
          'label': null,
          'link': 'https://google.com/application-guide'
        },
        ],
        'requiresFiles':false,
        'deleteReason':null,
      },
      {
        'id': 3,
        'name': 'Interpol Letter',
        'resources': [{
          'checklistItemId':3,
          'id':1,
          'label': null,
          'link': 'https://google.com/application-guide'
        },
        ],
        'requiresFiles':true,
        'deleteReason':null,
      }
    ]},
  {
    'destination': 'Lagos, Nigeria',
    checklist:[{'id': 8,
      'name': 'Visa Application',
      'resources': [{
        'checklistItemId':1,
        'id':2,
        'label': 'Application guide',
        'link': 'https://google.com/application-guide'
      },
      ],
      'requiresFiles':true,
      'deleteReason':null,
    },{
      'id': 9,
      'name': 'Yellow Fever',
      'resources': [{
        'checklistItemId':2,
        'id':1,
        'label': null,
        'link': 'https://google.com/application-guide'
      },
      ],
      'requiresFiles':true,
      'deleteReason':null,
    },
    {
      'id': 7,
      'name': 'Ticket Information',
      'resources': [{
        'checklistItemId':2,
        'id':1,
        'label': null,
        'link': 'https://google.com/application-guide'
      },
      ],
      'requiresFiles':false,
      'deleteReason':null,
    }
    ]}
];

export const mockRequest = {
  id: 'hg3456yh',
  name: 'Tester Demola',
  origin: 'Kampala',
  destination: 'New york',
  gender: 'Male',
  manager: 'Some manager',
  department: 'TDD',
  role: 'Senior Consultant',
  tripType: 'multi',
  picture: 'https://sgeeegege',
  trips: [
    {
      id: '123',
      origin: 'Nairobi',
      destination: 'Lagos, Nigeria',
      departureDate,
      returnDate,
    },
    {
      origin: 'New York',
      destination: 'Nairobi',
      departureDate,
      returnDate,
    },
    {
      id: '234',
      origin: 'New York',
      destination: 'Kampala, Uganda',
      departureDate,
      returnDate,
    }
  ],
};

// export default travelChecklists;
