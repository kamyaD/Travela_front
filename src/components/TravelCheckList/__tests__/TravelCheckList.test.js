import React from 'react';
import { shallow, mount } from 'enzyme';
import travelChecklistMockData from '../../../mockData/travelChecklistMockData';
import TravelCheckListPage from '../index';

describe('TravelChecklist Component', () => {
  let props = {
    request: {
      approver: 'John Doe',
      approverImage: 'xyz.jpeg',
      budgetApprovedAt: null,
      budgetApprovedBy: null,
      budgetStatus: 'Approved',
      comments: [],
      createdAt: '2019-03-09T22:45:33.286Z',
      deletedAt: null,
      department: 'Fellowship-Programs',
      gender: 'Male',
      id: 'c1Bwhq0KY',
      manager: 'John Doe',
      name: 'John Doe',
      picture: 'abc.jpg',
      role: 'Technical Team Lead',
      status: 'Approved',
      stipend: 123,
      timeApproved: '2019-03-09T22:45:54.887Z',
      travelCompletion: '25% complete',
      tripType: 'oneWay',
      trips: [{
        accommodationType: 'Hotel Booking',
        bedId: null,
        beds: null,
        checkInDate: null,
        checkOutDate: null,
        checkStatus: 'Not Checked In',
        createdAt: '2019-03-09T22:45:33.305Z',
        deletedAt: null,
        departureDate: '2019-06-01',
        destination: 'Nairobi, Kenya',
        id: 'Li3gOhUUCG',
        lastNotifyDate: null,
        notificationCount: 0,
        origin: 'Lagos, Nigeria',
        otherTravelReasons: null,
        reasons: {id: 3, title: 'nothing', description: '', 
          createdAt: '2019-02-28T22:46:28.714Z', updatedAt: '2019-02-28T22:46:28.714Z'},
        requestId: 'c1Bwhq0KY',
        returnDate: '2019-07-01',
        travelCompletion: 'false',
        travelReasons: 3,
        updatedAt: '2019-03-09T22:45:33.305Z',
      }]
    },
    modalType: '',
    openModal: jest.fn(),
    fetchSubmission: jest.fn(),
    postSubmission: jest.fn(),
    isFetching: false,
    showTravelChecklist: jest.fn(),
    closeModal: () => {},
    shouldOpen: false,
    userReadinessDocument: {},
    history: { push: jest.fn()},
    travelChecklists: {
      checklistItems: travelChecklistMockData,
      isLoading: false
    },
    submissionInfo: {
      submissions : [], 
      isUploading: [], 
      percentageCompleted: 80,
      itemsToCheck: [], 
      postSuccess: [], 
      tripType: 'return',
    },
    fileUploads: {
      cloudinaryUrl: '',
      error: '',
      isUploading: '',
      uploadSuccess: ''
    },
    uploadFile: jest.fn(),
    hideSubmit: null
  };

  const setup = (props) => mount(<TravelCheckListPage {...props} />);

  it ('should render the component', () => {
    const wrapper = setup(props);
    expect(wrapper).toMatchSnapshot();
    const travelChecklistRow = wrapper.find('.travelCheckList__row');
    const progressBarText2 = wrapper.find('.progressBar__text-head');
    const progressBarText1 = wrapper.find('.progressBar__text-foot');
    const travelChecklistCardHead = wrapper.find('.travelCheckList--card__head');
    const checklistItem = wrapper.find('.travelCheck-list');
    const progressBarMessage = wrapper.find('.progressBar__message');
    expect(travelChecklistRow.length).toBe(1);
    expect(progressBarText2.text()).toBe('You are');
    expect(progressBarText1.text()).toBe('Travel Ready');
    expect(progressBarMessage.text()).toBe('Complete the checklist to continue');
    expect(travelChecklistCardHead.length).toBe(2);
    expect(checklistItem.length).toBe(1);
  });
  
  it('should render submit button', () => {
    props.submissionInfo.percentageCompleted= 100;
    const wrapper = setup(props);
    const submitButton = wrapper.find('#submit-button');
    expect(submitButton.text()).toEqual('SUBMIT');
  });

  it('should call showTravelChecklist on componentWillRecieveProps', () => {
    const wrapper = setup(props);
    wrapper.setProps = {
      fileUploads: {
        uploadSuccess: '1234'
      }
    };
    expect(wrapper.instance().componentWillReceiveProps(props)).toBeCalled;
    expect(wrapper.instance().props.showTravelChecklist(props.request)).toBeCalled;
  });

  it('should uploadFile when handleFileUpload is called', ()=>{
    const wrapper = setup(props);
    wrapper.instance().handleFileUpload( {}, 1, 1, 1);
    expect(wrapper.instance().props.uploadFile).toBeCalled;
  });

  it('should open a modal when handleUserDocumentUpload is triggered', () => {
    const wrapper = setup(props);
    wrapper.instance().handleUserDocumentUpload('modal-1');
    expect(wrapper.instance().props.openModal(true, 'modal-1')).toBeCalled;
  });

  it('should return to requests page', () =>{
    const wrapper = setup(props);
    wrapper.instance().returnToRequestPage();
    expect(wrapper.props.history).toBeCalled;
  });

  it('should call showTravelChecklist when upload is successful', () => {
    const wrapper = setup(props);
    wrapper.setProps({
      fileUploads: {
        uploadSuccess: 'succesfully'
      },
      showTravelChecklist: jest.fn(),
      request: {}
    });
    expect(wrapper.instance().showTravelChecklist).toBeCalled;
  });


});
