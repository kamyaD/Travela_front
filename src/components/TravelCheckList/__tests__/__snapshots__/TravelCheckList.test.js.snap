// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TravelChecklist Component should render the component 1`] = `
<div
  className="travelCheck-list"
>
  <div
    className="travelCheckList__row mdl-grid"
  >
    <div
      className="travelCheckList__col-7 mdl-cell mdl-cell--7-col mdl-cell--5-col-tablet mdl-cell--12-col-phone"
    >
      <div
        className="travelCheckList--card"
      >
        <div
          className="travelCheckList--card__head"
        >
          These are the checklist items required to be submitted for this trip
        </div>
      </div>
    </div>
    <div
      className="travelCheckList__col-5 mdl-cell mdl-cell--5-col mdl-cell--3-col-tablet mdl-cell--12-col-phone "
    >
      <div
        className="travelCheckList--card travelCheckList--card__sticky"
      >
        <div
          className="travelCheckList--card__head"
        >
          Your Travel Readiness
        </div>
        <div
          className="progressBar"
        >
          <div
            className="progressBar__text"
          >
            <div
              className="progressBar__text-head"
            >
              You are
            </div>
            <div
              className="progressBar__text-foot"
            >
              Travel Ready
            </div>
          </div>
          <svg
            height={169}
            viewBox="0 0 169 169"
            width={169}
          >
            <circle
              className="circle-background"
              cx={84.5}
              cy={84.5}
              r={79.5}
              strokeWidth="10px"
            />
            <circle
              className="circle-progress"
              cx={84.5}
              cy={84.5}
              r={79.5}
              stroke="#3359DB"
              strokeWidth="10px"
              style={
                Object {
                  "strokeDasharray": 499.5132319207771,
                  "strokeDashoffset": 99.90264638415545,
                }
              }
              transform="rotate(-90 84.5 84.5)"
            />
            <text
              className="circle-text"
              dy=".3em"
              textAnchor="middle"
              x="50%"
              y="50%"
            >
              80%
            </text>
          </svg>
        </div>
        <div
          className="progressBar__message"
        >
          <p>
            Complete the checklist to continue
          </p>
        </div>
      </div>
      <div
        className="travelCheckList__row--button-area"
      >
        <div
          className="travelSubmission--submit-area__button"
        >
          <button
            className=""
            id="save-button"
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
